Improve parsing function and remove most parallel function

gutzbenj opened the pull request:
In order to speed things up with high resolution data, which was till now unusable, I changed the way parsing works. I replaced the multiple calls to `pandas.read_csv()` with one call and instead now the bytes are put together manually before loading station data into pandas. This also makes possible to remove whitespace in between which would usually have to be handled by pandas.

An additional implication that this change brought was the early removal of the "eor" column, as I was running into an parsing error with 10_minutes data where some files had "eor" while others didn't.

Additionally I removed the "parallel" options and instead made the download happen in parallel all the time, because why not? This way we also remove the keyword from several functions. Tests have changed marginally with this change.

With this change, I could improve loading time for 1_minute data (no broad test!) for one chosen station from 8 minutes to 10 seconds.

Please check it out!

amotl commented:
Thanks for your efforts improving the performance of the machinery!

amotl commented on wetterdienst/download/download.py:
```diff
@@ -3,7 +3,7 @@
 import zipfile
 from io import BytesIO
-from multiprocessing import Pool
+from multiprocessing.dummy import Pool
```
Shall we still use the "dummy" implementation or the real thing?

gutzbenj replied:
After experiencing the issues with real multicore processing I'd trend to try to prevent using it. Another con in my point of view is that for downloading and unpacking of the zip (which is the only current usage) threading should be enough effort to speed things up.

meteoDaniel commented:
@gutzbenj there are several reasons for not doing parallel download by default. In some cases this can crash your server blocking all CPU's or running into RAM issues. So this is a tool where you have to know what you are doing. 
But normally I would like to use it by default with the option to set it off? Maybe this is the most suitable way?

gutzbenj commented:
> @gutzbenj there are several reasons for not doing parallel download by default. In some cases this can crash your server blocking all CPU's or running into RAM issues. So this is a tool where you have to know what you are doing.
> But normally I would like to use it by default with the option to set it off? Maybe this is the most suitable way?

@meteoDaniel  So eventually my starting point with this discussion falsly claimed speed improvements. However the real issue finally was the parsing of datetimes. Nonetheless I still had faced problems running the original code under Python 3.8, which I hadn't seen before and those problems were related to multiprocessing. As of now I don't fully understand the reasons behind this error. The speed up in general only effects 1_minute and 10_minutes. While 10_minutes data parsing may circulate around upto 10 seconds for one station to complete, for 1 minute data this can stress up to 1 minute or eventually more. However as it simply fails under Python 3.8, I'd for now like to circumvent multiprocessing.

gutzbenj commented:
I again updated the downloading and parsing. It now realy relies on the parsing. I replaced the threading function in the download function by the new concurrent.futures. The tool uses multiple threads and is blazing fast (~1s) for 1 minute resolution. However parsing is still the bottleneck with casting the dtypes correctly, which I have simplified very much to get as little overhead as possible. Still, as multiprocessing is not running well on Windows and there is no obvious solution, I really don't want to set this option, besides the downside that the processor might be blocked. For station 3 from 1_minute historical precipitation (26 years of data) it takes now approx 1min20secs.

amotl commented:
LGTM!

meteoDaniel commented:
+1!

meteoDaniel commented on wetterdienst/download/download.py:
```diff
@@ -13,22 +13,18 @@
 
-def download_dwd_data(remote_files: List[str],
-                      parallel: bool = False) -> List[Tuple[str, BytesIO]]:
+def download_dwd_data(remote_files: List[str]) -> List[Tuple[str, BytesIO]]:
```
Name this _download_dwd_data_parallel
